database/src/base/rpc_server.rs

```diff
@@ -94,19 +94,19 @@ pub trait ReaderDbManager {
         &self,
         transaction_hash: &str,
         method_name: &str,
-    ) -> anyhow::Result<readnode_primitives::TransactionDetails>;
+    ) -> anyhow::Result<(u64, readnode_primitives::TransactionDetails)>;
```

Comment:
I agree, sometimes there might be a situation where returning a tuple is inevitable. Although, I don't believe this is such a situation.

Please consider introducing something like

```rust
struct ResponseWithBlockHeight<T> {
  block_height: u64,
  response: T,
}
```

Thus it'll be more maintainable in the future and less error-prone.

Reply:
We expected to rewrite the database crate. I left a TODO comment for the future.